cronjobs/create_pipeline_execution.py

```diff
@@ -0,0 +1,123 @@
+#!/bin/python3
+# -*- coding: utf-8 -*-
+
+# Algorithm to bring pipeline phases in an ideal order
+
+def split_to_equal_parts(lst, n):
+        for x in range(0, len(lst), n):
+            each_chunk = lst[x: n+x]
+            if len(each_chunk) < n:
+                each_chunk = each_chunk + [None for y in range(n-len(each_chunk))]
+            yield each_chunk
+
+def contains_elem(lst_2d, elem):
+    result = False
+    for lst in lst_2d:
+        if elem in lst:
+            result = True
+    return result
+
+def min_length_list(input_list):
+    #min_length = min(len(x) for x in input_list )
+    min_list = min(input_list, key = len)
+    min_pos = input_list.index(min_list)
+    return(min_pos, min_list)
+
+def Main():
+
+    # Number of parallel FIFO groups
+    number_of_groups = 2 # = 1 without parallelization
+    # Number of FIFO phases in a FIFO group
+    items_in_group = 3
+    phases = ["3","4","5","6","14","7","8","9","13"]
+    # Dependency list has to be built dynamically later
+    # e.g. ["5","3"] means that "5" can only be run after "3"
+    deps = [["5","3"],["7","4"],["8","5"],["9","7"],["6","13"],["3","13"],["b","a"],["c","a"],["m","k"],["z","x"],["z","k"],["c","b"]]
```

Comment:
Sometimes phases have multiple dependencies. For example, have a look at ixtheo_marc_pipeline.sh:

```
StartPhase "Add Keyword Synonyms from Authority Data"
(add_synonyms \
    GesamtTiteldaten-post-phase"$((PHASE-1))"-"${date}".mrc \
    Normdaten-partially-augmented2-"${date}".mrc \
    GesamtTiteldaten-post-phase"$PHASE"-"${date}".mrc >> "${log}" 2>&1 && \
EndPhase || Abort) &
wait
```
Maybe it would be better to use a more complex datatype, here's an example of a hybrid list (there might be better options!):
`deps = [["add_synonyms", ["map_ddc_to_ixtheo_notations", "add_authority_beacon_information"]]]`